src/responder/pam/pamsrv_cmd.c

```diff
@@ -1418,6 +1418,15 @@ void pam_reply(struct pam_auth_req *preq)
         goto done;
     }
 
+#if BUILD_PASSKEY
+    if(pd->cmd == SSS_PAM_AUTHENTICATE &&
+       pd->pam_status == PAM_NEW_AUTHTOK_REQD &&
```

Comment:
How / when does it happen?

Reply:
This is needed when the IPA user has auth types `passkey + password` set, and  has an expired password. Without this code entering the passkey PIN causes the following:

```
justin@fedora:~$ su - testuser1
Insert your passkey device, then press ENTER.
Enter PIN:
Password expired. Change your password now.
Current Password: 
```

The krb5_child pre-auth returns `Password has expired`, but here we are authenticating with Passkey authtok type (**SSS_AUTHTOK_TYPE_PASSKEY_REPLY**) therefore we can return PAM_SUCCESS as we don't need to return  **PAM_NEW_AUTHTOK_REQD** to require a password change.
```

justin@fedora:~$ su - testuser1
Insert your passkey device, then press ENTER.
Enter PIN:
Last login: Tue Mar  5 11:57:03 EST 2024 on pts/11
testuser1@fedora:~$ 
```

Reply:
Thank you for the explanation.